File: 04-copies-of-teaching-material/week04/ros2_lab/handson/my_turtle_circle/circle.py
import time
import rclpy
from rclpy.node import Node
from rclpy.signals import SignalHandlerOptions
from geometry_msgs.msg import Twist

class Circle(Node):
    def __init__(self):
        super().__init__('circle')
        self.pub = self.create_publisher(Twist, '/turtle1/cmd_vel', 10)
        self.create_timer(0.1, self.send)

    def send(self):
        msg = Twist()
        msg.linear.x = 2.0
        msg.angular.z = 1.0
        self.pub.publish(msg)

def main():
    rclpy.init(signal_handler_options=SignalHandlerOptions.NO)
    node = Circle()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.pub.publish(Twist())
        time.sleep(0.3)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: 04-copies-of-teaching-material/week04/ros2_lab/handson/ros2_motion_cpp/circle.cpp
#include <chrono>
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

using namespace std::chrono_literals;

class Circle : public rclcpp::Node
{
public:
  Circle() : Node("circle_cpp")
  {
    pub_ = create_publisher<geometry_msgs::msg::Twist>("/turtle1/cmd_vel", 10);
    timer_ = create_wall_timer(100ms, [this]() {
      geometry_msgs::msg::Twist msg;
      msg.linear.x = 2.0;
      msg.angular.z = 1.0;
      pub_->publish(msg);
    });
  }

private:
  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr pub_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Circle>());
  rclcpp::shutdown();
  return 0;
}
